CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/SplitNodesPlugin.java

```diff
@@ -130,82 +144,132 @@ public void updateParameters(Graph graph, PluginParameters parameters) {
+
+        int vertexSelectedAttributeId = graph.getAttribute(GraphElementType.VERTEX, VisualConcept.VertexAttribute.SELECTED.getName());
+        int vertexIdentifierAttributeId = graph.getAttribute(GraphElementType.VERTEX, VisualConcept.VertexAttribute.IDENTIFIER.getName());
+        List<Integer> newVertices = new ArrayList<>();
+
+        final int graphVertexCount = graph.getVertexCount();
+        for (int position = 0; position < graphVertexCount; position++) {
+            int currentVertexId = graph.getVertex(position);
+
+            if (graph.getBooleanValue(vertexSelectedAttributeId, currentVertexId)) {
+                final String identifier = graph.getStringValue(vertexIdentifierAttributeId, currentVertexId);
+
+                if (identifier != null && identifier.contains(character) && identifier.indexOf(character) < identifier.length() - character.length()) {
+                    String leftNodeIdentifier = "";
+                    if (allOccurrences) {
+                        final List<String> substrings = new ArrayList<>(Arrays.asList(identifier.split(character)));
+                        leftNodeIdentifier = substrings.get(0);
+                        for (int i = 1; i < substrings.size(); i++) {
+                            newVertices.add(createNewNode(graph, position, substrings.get(i), linkType, splitIntoSameLevel));
+                        }
 
-        query.reset();
-        while (query.next()) {
-            final String identifier = query.get(SOURCE_IDENTIFIER);
-            if (identifier != null && identifier.contains(character) && identifier.indexOf(character) < identifier.length() - character.length()) {
-                if (allOccurrences) {
-                    final List<String> substrings = new ArrayList<>(Arrays.asList(identifier.split(character)));
-                    final String left = substrings.get(0);
-                    for (int i = 1; i < substrings.size(); i++) {
-                        final String right = substrings.get(i);
-                        editResultStore(result, left, right, query, linkType);
+                    } else {
+                        final int i = identifier.indexOf(character);
+                        leftNodeIdentifier = identifier.substring(0, i);
+                        newVertices.add(createNewNode(graph, position, identifier.substring(i + 1), linkType, splitIntoSameLevel));
                     }
-                } else {
-                    final int i = identifier.indexOf(character);
-                    final String left = identifier.substring(0, i);
-                    final String right = identifier.substring(i + 1);
-                    editResultStore(result, left, right, query, linkType);
+                    // Rename the selected node
+                    graph.setStringValue(vertexIdentifierAttributeId, currentVertexId, leftNodeIdentifier);
                 }
             }
         }
+        if (!newVertices.isEmpty()) {
```

Comment:
Added it to avoid resetting the view when there's no matching node to split. I can remove it if you think it's okay to just flicker the graph sometimes.